### PR TITLE
DR-161 Implement delete response

### PR DESCRIPTION
**Pull request by ddietterich**
- added the delete response model to the swagger
- changed study and dataset delete flights to return the response
- factored out common response processing code for JobService and StudyService into FlightUtils
- fixed tests to process with the new code

### CODE REVIEW

**Review by thathert — APPROVED** [review]
LGTM

File: src/main/java/bio/terra/service/StudyService.java [hunk 1]
```diff
@@ -35,6 +37,11 @@ public StudyService(StudyDao studyDao, Stairway stairway, DatasetService dataset
         this.datasetService = datasetService;
     }
 
+    /**
+     * Create a study
+     * @param studyRequest from the REST API
+     * @return a StudySummaryModel
+     */
```
Comment (ahaessly) [inline src/main/java/bio/terra/service/StudyService.java]: I'm of the opinion that we should use names that are self documenting so we don't need boiler plate documentation. The swagger is the documentation for the cli. We should only add documentation when the code is doing something surprising so when documenting additional functionality that needs to be added. I guess we need a team discussion.

File: src/main/resources/data-repository-openapi.yaml [hunk 2]
```diff
@@ -267,20 +297,26 @@ paths:
       parameters:
       - $ref: '#/parameters/Id'
       responses:
-        202: # kicks off a delete study job
-          description: Study deletion job started & url for polling is in the response header
-          headers:
-            location:
-              type: string
-              description: url for the job poller
+        200:
```
Comment (ahaessly) [inline src/main/resources/data-repository-openapi.yaml:300]: Are we just changing this to synchronous temporarily? If so, can we add a comment to that affect?

**Review by RoriCremer — APPROVED** [review]

**Review by ahaessly — APPROVED** [review]
a couple of comments

File: src/main/java/bio/terra/service/StudyService.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/StudyService.java]: Yes, I agree. I added this in when I had more to say and didn't remove it when I had nothing to say. I'll remove.

File: src/main/resources/data-repository-openapi.yaml (hunk 2 above)
Comment (ddietterich) [reply src/main/resources/data-repository-openapi.yaml:300]: As we discussed, we will keep study create and delete synchronous for now, so I removed the 202.

**Pull request merged**